packages/@lightningjs/ui-components/src/components/FocusManager/FocusManager.d.ts

```diff
@@ -19,44 +19,129 @@
+  lng.Component.NewPatchTemplate<lng.Component.Constructor> | lng.Component
+>;
+declare namespace FocusManager {
+  export interface TemplateSpec extends Base.TemplateSpec {
+    /**
+     * the navigation direction
+     */
+    direction?: NavigationDirectionType;
+    /**
+     * child element or elements of the FocusManager
+     */
     items: Array<
-      lng.Component.NewPatchTemplate<lng.Component.Constructor> | lng.Component
-    >
-  );
-  get items(): Array<
-    lng.Component.NewPatchTemplate<lng.Component.Constructor> | lng.Component
-  >;
+      | lng.Component.NewPatchTemplate<lng.Component.Constructor>
+      | typeof lng.Component
+    >;
+    /**
+     * index of currently selected item
+     */
+    selectedIndex?: number;
+
+    /**
+     * enables wrapping behavior, so `selectNext` selects the first item if the current item is the last on the list and vice versa
+     */
+    wrapSelected?: boolean;
+  }
+}
+
+declare class FocusManager<
+  TemplateSpec extends FocusManager.TemplateSpec,
+  TypeConfig extends lng.Component.TypeConfig
+> extends Base<TemplateSpec, TypeConfig> {
+  // Properties
+
+  /**
+   * the navigation direction
+   */
+  direction?: NavigationDirectionType;
+
+  /**
+   * child element or elements of the FocusManager
+   */
+  items: FocusItemsType;
+
+  /**
+   * index of currently selected item
+   */
+  selectedIndex: number;
+
+  /**
+   * enables wrapping behavior, so `selectNext` selects the first item if the current item is the last on the list and vice versa
+   */
+  wrapSelected: boolean;
+
+  // Accessors
+
   get Items(): lng.Element;
 
+  /**
+   * returns the currently selected component
+   */
   get selected(): lng.Component;
-  get selectedIndex(): number;
-  set selectedIndex(index: number);
+
+  /**
+   * returns an array containing the children of the FocusManager that are fully within the visible bounds of the FocusManager
+   */
   get fullyOnScreenItems(): Array<lng.Component>;
 
-  appendItems(
-    items: Array<lng.Component.NewPatchTemplate<lng.Component.Constructor>>
-  ): void;
-  appendItemsAt(
-    items: Array<lng.Component.NewPatchTemplate<lng.Component.Constructor>>,
-    idx: number
-  ): void;
-  prependItems(
-    items: Array<lng.Component.NewPatchTemplate<lng.Component.Constructor>>
-  ): void;
+  // Methods
+
+  /**
+   * adds the provided Rows to the end of the FocusManager children
+   * @param items array of Rows or Row-type patch objects
+   */
+  appendItems(items: FocusItemsType): void;
+
+  /**
+   * adds the provided Rows to the end of the FocusManager children at the provided index
+   * @param items array of Rows or Row-type patch objects
+   * @param idx index to insert rows at
+   */
+  appendItemsAt(items: FocusItemsType, idx: number): void;
+
+  /**
+   * adds the provided Rows to the beginning of the FocusManager children
+   * @param items array of Rows or Row-type patch objects
+   */
+  prependItems(items: FocusItemsType): void;
```

Comment:
Are these always Rows or Row-type objects?

I noticed that your comment for `removeItemAt` simply refers to "Item" instead.